Any file is taking seconds to open

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Since Zed 0.162.x version, It started taking seconds to open any file. Trying to debug, i ran

```bash
tail -f ~/Library/Logs/Zed/Zed.log
```

and while waiting to file open, this was logged:

```
2024-12-02T11:43:50.147601-03:00 [INFO] Env ("/Users/ggalves/.asdf/shims/python") in Path resolved as ResolvedPythonEnv { executable: "/Users/ggalves/.asdf/installs/python/3.11.8/bin/python", prefix: "/Users/ggalves/.asdf/installs/python/3.11.8", version: "3.11.8.final.0", is64_bit: true, symlinks: Some(["/Users/ggalves/.asdf/installs/python/3.11.8/bin/python", "/Users/ggalves/.asdf/shims/python"]) } and reported as Some(GlobalPaths)
2024-12-02T11:43:50.149346-03:00 [INFO] Env ("/Users/ggalves/.asdf/shims/python3") in Path resolved as ResolvedPythonEnv { executable: "/Users/ggalves/.asdf/installs/python/3.11.8/bin/python3", prefix: "/Users/ggalves/.asdf/installs/python/3.11.8", version: "3.11.8.final.0", is64_bit: true, symlinks: Some(["/Users/ggalves/.asdf/installs/python/3.11.8/bin/python3", "/Users/ggalves/.asdf/shims/python3"]) } and reported as Some(GlobalPaths)
2024-12-02T11:43:50.735969-03:00 [ERROR] Python Execution for "/Users/ggalves/.asdf/shims/python3.10" produced an output "" without a separator
2024-12-02T11:43:50.736349-03:00 [WARN] Unknown Python Env "/Users/ggalves/.asdf/shims/python3.10"
2024-12-02T11:43:50.737371-03:00 [INFO] Env ("/Users/ggalves/.asdf/shims/python3.11") in Path resolved as ResolvedPythonEnv { executable: "/Users/ggalves/.asdf/installs/python/3.11.8/bin/python3.11", prefix: "/Users/ggalves/.asdf/installs/python/3.11.8", version: "3.11.8.final.0", is64_bit: true, symlinks: Some(["/Users/ggalves/.asdf/installs/python/3.11.8/bin/python3.11", "/Users/ggalves/.asdf/shims/python3.11"]) } and reported as Some(GlobalPaths)
2024-12-02T11:43:51.136441-03:00 [ERROR] Python Execution for "/Users/ggalves/.asdf/shims/python3.12" produced an output "" without a separator
2024-12-02T11:43:51.136823-03:00 [WARN] Unknown Python Env "/Users/ggalves/.asdf/shims/python3.12"
2024-12-02T11:43:51.508586-03:00 [ERROR] Python Execution for "/Users/ggalves/.asdf/shims/python3.8" produced an output "" without a separator
2024-12-02T11:43:51.508781-03:00 [WARN] Unknown Python Env "/Users/ggalves/.asdf/shims/python3.8"
2024-12-02T11:43:51.862024-03:00 [ERROR] Python Execution for "/Users/ggalves/.asdf/shims/python3.9" produced an output "" without a separator
2024-12-02T11:43:51.862205-03:00 [WARN] Unknown Python Env "/Users/ggalves/.asdf/shims/python3.9"
```

Note: I was not trying to open a python file. This problem happens in every file that there is no open tab

Seems to be a problem with this auto detection of Python Env. As noted from logs, I do use python from asdf version manager

### Environment

Zed: v0.163.2 (Zed)
OS: macOS 14.6.1
Memory: 18 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

_No response_

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary>

<!-- Click below this line and paste or drag-and-drop your log-->
```
2024-12-02T11:43:50.147601-03:00 [INFO] Env ("/Users/ggalves/.asdf/shims/python") in Path resolved as ResolvedPythonEnv { executable: "/Users/ggalves/.asdf/installs/python/3.11.8/bin/python", prefix: "/Users/ggalves/.asdf/installs/python/3.11.8", version: "3.11.8.final.0", is64_bit: true, symlinks: Some(["/Users/ggalves/.asdf/installs/python/3.11.8/bin/python", "/Users/ggalves/.asdf/shims/python"]) } and reported as Some(GlobalPaths)
2024-12-02T11:43:50.149346-03:00 [INFO] Env ("/Users/ggalves/.asdf/shims/python3") in Path resolved as ResolvedPythonEnv { executable: "/Users/ggalves/.asdf/installs/python/3.11.8/bin/python3", prefix: "/Users/ggalves/.asdf/installs/python/3.11.8", version: "3.11.8.final.0", is64_bit: true, symlinks: Some(["/Users/ggalves/.asdf/installs/python/3.11.8/bin/python3", "/Users/ggalves/.asdf/shims/python3"]) } and reported as Some(GlobalPaths)
2024-12-02T11:43:50.735969-03:00 [ERROR] Python Execution for "/Users/ggalves/.asdf/shims/python3.10" produced an output "" without a separator
2024-12-02T11:43:50.736349-03:00 [WARN] Unknown Python Env "/Users/ggalves/.asdf/shims/python3.10"
2024-12-02T11:43:50.737371-03:00 [INFO] Env ("/Users/ggalves/.asdf/shims/python3.11") in Path resolved as ResolvedPythonEnv { executable: "/Users/ggalves/.asdf/installs/python/3.11.8/bin/python3.11", prefix: "/Users/ggalves/.asdf/installs/python/3.11.8", version: "3.11.8.final.0", is64_bit: true, symlinks: Some(["/Users/ggalves/.asdf/installs/python/3.11.8/bin/python3.11", "/Users/ggalves/.asdf/shims/python3.11"]) } and reported as Some(GlobalPaths)
2024-12-02T11:43:51.136441-03:00 [ERROR] Python Execution for "/Users/ggalves/.asdf/shims/python3.12" produced an output "" without a separator
2024-12-02T11:43:51.136823-03:00 [WARN] Unknown Python Env "/Users/ggalves/.asdf/shims/python3.12"
2024-12-02T11:43:51.508586-03:00 [ERROR] Python Execution for "/Users/ggalves/.asdf/shims/python3.8" produced an output "" without a separator
2024-12-02T11:43:51.508781-03:00 [WARN] Unknown Python Env "/Users/ggalves/.asdf/shims/python3.8"
2024-12-02T11:43:51.862024-03:00 [ERROR] Python Execution for "/Users/ggalves/.asdf/shims/python3.9" produced an output "" without a separator
2024-12-02T11:43:51.862205-03:00 [WARN] Unknown Python Env "/Users/ggalves/.asdf/shims/python3.9"
```
<!-- Click above this line and paste or drag-and-drop your log--></details>

--- a/crates/language/src/toolchain.rs
+++ b/crates/language/src/toolchain.rs
@@ -24,7 +24,7 @@ pub struct Toolchain {
     pub as_json: serde_json::Value,
 }
 
-#[async_trait(?Send)]
+#[async_trait]
 pub trait ToolchainLister: Send + Sync {
     async fn list(
         &self,

--- a/crates/languages/src/python.rs
+++ b/crates/languages/src/python.rs
@@ -536,7 +536,7 @@ fn env_priority(kind: Option<PythonEnvironmentKind>) -> usize {
     }
 }
 
-#[async_trait(?Send)]
+#[async_trait]
 impl ToolchainLister for PythonToolchainProvider {
     async fn list(
         &self,

--- a/crates/project/src/toolchain_store.rs
+++ b/crates/project/src/toolchain_store.rs
@@ -312,11 +312,12 @@ impl LocalToolchainStore {
                 .ok()?
                 .await;
             let language = registry.language_for_name(&language_name.0).await.ok()?;
-            let toolchains = language
-                .toolchain_lister()?
-                .list(root.to_path_buf(), project_env)
-                .await;
-            Some(toolchains)
+            let toolchains = language.toolchain_lister()?;
+            Some(
+                cx.background_executor()
+                    .spawn(async move { toolchains.list(root.to_path_buf(), project_env).await })
+                    .await,
+            )
         })
     }
     pub(crate) fn active_toolchain(

--- a/crates/project/src/toolchain_store.rs
+++ b/crates/project/src/toolchain_store.rs
@@ -311,13 +311,14 @@ impl LocalToolchainStore {
                 })
                 .ok()?
                 .await;
-            let language = registry.language_for_name(&language_name.0).await.ok()?;
-            let toolchains = language.toolchain_lister()?;
-            Some(
-                cx.background_executor()
-                    .spawn(async move { toolchains.list(root.to_path_buf(), project_env).await })
-                    .await,
-            )
+
+            cx.background_executor()
+                .spawn(async move {
+                    let language = registry.language_for_name(&language_name.0).await.ok()?;
+                    let toolchains = language.toolchain_lister()?;
+                    Some(toolchains.list(root.to_path_buf(), project_env).await)
+                })
+                .await
         })
     }
     pub(crate) fn active_toolchain(

